### ISSUE TITLE
Failed to resolve component / Invalid value used as weak map key

### ISSUE THREAD

**Issue by Extarys** [inkline/nuxt-module-inkline#4]
I'm using the latest Nuxt 3 package as of today (I also had this error with my previous (unknown) version.

**Console error:**
```
[Vue warn]: Failed to resolve component: i-layout
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Failed to resolve component: i-layout-aside
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Failed to resolve component: i-layout-header
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Failed to resolve component: i-hamburger-menu
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Failed to resolve component: i-layout-content
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
[Vue warn]: Failed to resolve component: i-layout-footer
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
Invalid value used as weak map key
  at WeakMap.set (<anonymous>)  
  at normalizePropsOptions (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3384:11)  
  at createComponentInstance (./node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6851:23)  
  at renderComponentVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:197:22)  
  at Module.ssrRenderComponent (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:629:12)  
  at _sfc_ssrRender (file://./.nuxt/dist/server/server.mjs:2160:31)  
  at renderComponentSubTree (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:269:13)  
  at renderComponentVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:214:16)  
  at renderVNode (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:310:22)  
  at renderComponentSubTree (./node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:275:13)
```

Webpage throws a 500.

**packages.js:**
```
  "devDependencies": {
    "@vueuse/nuxt": "^7.6.0",
    "nuxt-windicss": "^2.2.4",
    "nuxt3": "^3.0.0-27430570.4cfad4d",
    "sass": "^1.49.9"
  },

  "dependencies": {
    "@gun-vue/components": "^0.7.4",
    "@gun-vue/composables": "^0.7.4",
    "@gun-vue/relay": "^0.3.5",
    "@inkline/inkline": "^3.0.8",
    "@inkline/nuxt": "^3.0.5",
    "@pinia/nuxt": "^0.1.8",
    "daisyui": "^1.25.4", // < Was planning on using this before discovering Inkline
    "gun": "^0.2020.1235",
    "pinia": "^2.0.11"
  }
```

**nuxt.config.ts:**
```
export default defineNuxtConfig({
	buildModules: [
		'@vueuse/nuxt',
		'@pinia/nuxt',
		// 'nuxt-windicss',
		'@inkline/nuxt'
	],

	build: {
		// transpile: ['@gun-vue/composables']
	},

	vueuse: {
		ssrHandlers: true,
	},

	inkline: {
        // Plugin options (optional)
    }
})
```

---
Thanks for this, I'm very excited to start using it :)

**Comment by Extarys** [comment]
@alexgrozav Yesss thank you very much! <3

I know Nuxt 3 is still being worked on so it's normal that some stuff breaks from time to time :D

**Comment by Extarys** [comment]
I think Nuxt 3 broke inkline, again :face_exhaling:  I just got started playing with it :angry: ahahah. Hopefully Nuxt3 will be more stable soon :tm: 

No rush - I would fix it but I have no idea what's going on :joy: Thanks for your hard work btw, I hope Inkline gets more attention from the community as I hate that you're all alone doing this.

"nuxt3": "^3.0.0-27439932.78761c2",
"@inkline/inkline": "^3.0.8",
"@inkline/nuxt": "^3.0.5",

**Comment by alexgrozav** [comment]
Hey @Extarys! I just tried a clean installation and I'm not getting any errors. 

Also tried with your versions. Can you try removing `package-lock.json`, `.nuxt` and `node_modules`, and then run `npm install` again?

**Comment by titan-65** [comment]
'''
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
(node:44315) UnhandledPromiseRejectionWarning: TypeError: Invalid value used as weak map key
    at WeakMap.set (<anonymous>)
    at normalizePropsOptions (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3384:11)
    at createComponentInstance (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:6851:23)
    at renderComponentVNode (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:197:22)
    at Module.ssrRenderComponent (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:629:12)
    at ssrRender (file:///Users/vantolbennett/Developer/2022/nuxt3-church-code/.nuxt/dist/server/server.mjs:12143:31)
    at renderComponentSubTree (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:269:13)
    at renderComponentVNode (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:214:16)
    at renderVNode (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:310:22)
    at renderVNodeChildren (/Users/vantolbennett/Developer/2022/nuxt3-church-code/node_modules/@vue/server-renderer/dist/server-renderer.cjs.js:325:9)
(node:44315) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
'''
Good day, i am the same issue here with this bug, everything shows up on screen thou however my terminal looks like that above. I am liking this UI keep the good work thou! I have used with regular Vue projects.

**Comment by schoenix** [comment]
I had a similar warning (It rendered the component from my plugin correctly but leaving this warning). In the browser console I had a different warning (something with hydration) which lead me to the solution.

The plugin wanted to be rendered client-side only (I had another error with document not found before), so I added the .client.ts to the name of the plugin.

I created a site (index.vue in my case). But I had to surround the NuxtPage tag in the app.vue file with the ClientOnly tags to get rid of this warning, because it wanted to render the index.vue site on the server, but with a tag that is client-only the type is not known at this time.

I hope this might help a little bit.
